CedarJava/src/main/java/com/cedarpolicy/model/slice/Policy.java

```diff
@@ -51,4 +65,38 @@ public Policy(
     public String toString() {
         return "// Policy ID: " + policyID + "\n" + policySrc;
     }
+
+    public static Policy parseStaticPolicy(String policyStr) throws InternalException, NullPointerException {
+        var policyText = parsePolicyJni(policyStr);
+        return new Policy(policyText, null);
+    }
+
+    public static Policy parsePolicyTemplate(String templateStr)  throws InternalException, NullPointerException {
+        var templateText = parsePolicyTemplateJni(templateStr);
+        return new Policy(templateText, null);
+    }
+
+    /**
+     * This method takes in a Policy and a list of Instantiations and calls Cedar JNI to ensure those slots
+     * can be used to instantiate the template. If the Template is validated ahead of time by using Policy.parsePolicyTemplate
+     * and the Instantiations are also ensured to be valid (for example, by validating their parts using EntityTypeName.parse
+     * and EntityIdentifier.parse), then this can only fail because the slots in the template don't match the instantiations.
+     * Hence we return false, since there's only one failure reason that can't be determined from other means.
+     * @param p Policy object constructed from a valid template. Best if built from Policy.parsePolicyTemplate
+     * @param principal EntityUid to put into the principal slot. Leave null if there's no principal slot
+     * @param resource EntityUid to put into the resource slot. Leave null if there's no resource slot
+     * @return
+     */
+    public static boolean validateTemplateLinkedPolicy(Policy p, EntityUID principal, EntityUID resource) {
+        try {
+            return validateTemplateLinkedPolicyJni(p.policySrc, principal, resource);
+        } catch (Exception e) {
```

Comment:
Do we want to swallow every exception here? Internal exceptions might not mean an invalid policy.

Reply:
What else could they mean? Bear in mind that the principal, resource, and policy have already been parsed by the library so they're valid

Reply:
I'm more thinking of exceptions that don't arise from the policy parsing calls specifically. For example, if we have a bug in our JNI code, perhaps that would surface as an exception from the JVM that we would want to bubble up. Possibly worrying too much here, but if we have a base class for our exceptions sitting around (`InternalException`?) then it might make sense to use that.

Reply:
I think im convinced to make it throw